suse2013/common/trim-verbatim.xsl

```diff
@@ -0,0 +1,94 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+    Purpose:
+     Trim start/end of verbatim areas (screen, programlisting, synopsis)
+
+
+    Author:     Stefan Knorr <sknorr@suse.de>
+    Copyright:  2015
+
+-->
+<xsl:stylesheet version="1.0"
+  xmlns:xsl="http://www.w3.org/1999/XSL/Transform">
+
+<xsl:template match="programlisting/text()|screen/text()|synopsis/text()">
+  <xsl:choose>
+    <xsl:when test="not(preceding-sibling::node()) or preceding-sibling::processing-instruction()[not(preceding-sibling::node())]">
+      <xsl:call-template name="trim-verbatim-whitespace-start"/>
+    </xsl:when>
+    <xsl:when test="not(following-sibling::node()) or following-sibling::processing-instruction()[not(following-sibling::node())]">
+      <xsl:call-template name="trim-verbatim-whitespace-end"/>
+    </xsl:when>
+    <xsl:otherwise>
+      <xsl:value-of select="self::text()"/>
+    </xsl:otherwise>
+  </xsl:choose>
+</xsl:template>
+
+<xsl:template name="trim-verbatim-whitespace-start">
+  <!-- Goal: At the start of text, trim lines or that are empty or contain only
+       spaces.
+       Do not trim lines that contain spaces at the beginning and then continue
+       with text. Doing that might destroy some content: e.g. command-line
+       output formatted as a table. -->
+  <xsl:param name="input" select="self::text()"/>
```

Comment:
Would it not rather be `self::node()` if I wanted to be more generic? I am already in a text node, so doing `text()` makes little sense, because that would translate to `self::node()/child::text()`.